### ISSUE TITLE
Release plan for 1.1 and 2.0

### ISSUE THREAD

**Issue by paazmaya** [casperjs/casperjs#1177]
I would like to have a bit of clarity in order to have my focus targeted on certain issues.
Currently there are Milestones for `1.1` and `2.0` which both have issues marked for them.
Also there are plenty of issues that have no milestone, but might need to be fixed before either of those releases can come true.

What I have gathered from the comments, regarding PhantomJS version parity, is that the release `1.1` would support PhantomJS `1.9.x` and release `2.0` would be the starting point for supporting PhantomJS `2.0`.

What is not fully clear to me, is that will the support for PhantomJS `1.9.x` be preserved in CasperJS `2.0`?

I would like to have the release `1.1` to be released soon, so that the focus for PhantomJS `2.0`, thus release `2.0` could be increased.

@n1k0 are the milestones up to date, and if not, would it be possible to have them reviewed and "locked" so that when the issues are fixed, the releases can be done?

Once there is a plan, it can be written in the documentation and users can be aware of it.

**Comment by n1k0** [comment]
> @n1k0 are the milestones up to date

Not at all.

> if not, would it be possible to have them reviewed and "locked" so that when the issues are fixed, the releases can be done?

I don't have the required time to handle this workload properly (nor have the required energy as well tbh); would you want to drive this — in collaboration with @mickaelandrieu?

The idea would be to triage issues & PRs and ensure they're all qualified and if so that the proper milestone is assigned — you would decide. For PRs, code could be reviewed by both of you, and I could be pinged only if you disagree on some particular change. Then once everything is triaged and PRs are landed, I can take care of releasing new versions.

Would this work for you guys?

Note: I'll try to initiate project ownership transfer to people having more time and active usage of it than me asap, I've been a really bad maintainer for the past year.

**Comment by mickaelandrieu** [comment]
@n1k0 for me it's ok.

What I suggest for the next 1.1 release is:
* fix critical bug if exists > issues/PR usage

What I suggest for the next 2.0-dev is:
* casperjs2 should be an npm module
* casperjs2 should be compatible with phantomjs >2 & < 3
* casperjs test api should be move on this own repository, and we should work on a better integration of libraries like mocha: in others words, casperjs2 shouldn't be a "test library" anymore

This is a lot of work and I think that we can probably release 1.1 "stable" soon, and start 2.x-dev release now with a target of 2016 for a first 2.0 "stable" release.

Note this is only suggests, and I will be very happy to accepts pros and cons about it, and also consider totaly differents suggests :)

**Comment by paazmaya** [comment]
@n1k0 ok for me too.
@mickaelandrieu sounds great, splitting into modules and use of npm

Perhaps it would be the most clear solution to create a `v1.1`branch which would serve only as preparation for the `1.1` release and once done, perhaps maintain few bug fixed, namely `1.1.1` and so forth.
Master branch would then start going towards `2.0`, as the core `casperjs` modules, while the test framework could move to `casperjs-test` (or something similar).

All of this could be under the GitHub organisation called...? casperjs.

I will start looking at milestones and putting issues in/out. In case either/anyone sees something that is out of place, please let me know.

**Comment by mickaelandrieu** [comment]
@paazmaya :+1: totaly agree on this point,
* 1.1 should be moved to his own branch
* master should reflect 2.0 branch and be able to accept BC.

Let's take 2 or 3 weeks to qualify issues and pull requests to decide what will be the 1.1, and move all others to the 2.0 milestone.

**Comment by paazmaya** [comment]
Sounds good!

**Comment by istr** [comment]
> Let's take 2 or 3 weeks to qualify issues and pull requests to decide what will be the 1.1, and move all others to the 2.0 milestone.

See [my 2 cents in depth here](https://github.com/n1k0/casperjs/pull/1352#issuecomment-159935751).

tl;dr
- 1.0.* *old and busted (obsolete)*: supports phantomjs up to 1.8.* (< 1.9.0)
- 1.1.* *next/current stable*: supports phantomjs 1.9.1 and later as well as slimerjs (experimental), we are only a few steps away from that (as soon as possible, Q1/2016 the latest)
- 2.0.* *new hotness*: will support phantomjs 2 and later, possibly breaks API with 1.* versions, comes with fancy new API and modular structure -- let's see if there is enough interest / manpower to really tackle this before diving deep into it

**Comment by BIGjuevos** [comment]
Pretty sure this is obsoleted by the new 2.0 thread and the fact that 1.1 has been released.
